packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -159,20 +159,53 @@ List<String> _xcodeBuildSettingsLines({
   }
 
   final String buildNameToParse = buildInfo?.buildName ?? project.manifest.buildName;
-  final String buildName = validatedBuildNameForPlatform(TargetPlatform.ios, buildNameToParse);
-  if (buildName != null) {
-    xcodeBuildSettings.add('FLUTTER_BUILD_NAME=$buildName');
+  final bool buildNameIsMissing = buildNameToParse == null || buildNameToParse.isEmpty;
+  String buildName;
+  const String defaultBuildName = '1.0.0';
+  if (buildNameIsMissing) {
+    buildName = defaultBuildName;
+  } else {
+    buildName = validatedBuildNameForPlatform(TargetPlatform.ios, buildNameToParse);
   }
 
-  String buildNumber = validatedBuildNumberForPlatform(TargetPlatform.ios, buildInfo?.buildNumber ?? project.manifest.buildNumber);
-  // Drop back to parsing build name if build number is not present. Build number is optional in the manifest, but
-  // FLUTTER_BUILD_NUMBER is required as the backing value for the required CFBundleVersion.
-  buildNumber ??= validatedBuildNumberForPlatform(TargetPlatform.ios, buildNameToParse);
+  final String buildNumberToParse = buildInfo?.buildNumber ?? project.manifest.buildNumber;
+  final bool buildNumberIsMissing =
+      (buildNumberToParse == null || buildNumberToParse.isEmpty) && buildNameIsMissing;
+  String buildNumber;
+
+  const String defaultBuildNumber = '1';
+  if (buildNumberIsMissing) {
+    buildNumber = defaultBuildNumber;
+  } else {
+    buildNumber = validatedBuildNumberForPlatform(TargetPlatform.ios, buildNumberToParse);
+    // Drop back to parsing build name if build number is not present. Build number is optional in the manifest, but
+    // FLUTTER_BUILD_NUMBER is required as the backing value for the required CFBundleVersion.
+    buildNumber ??= validatedBuildNumberForPlatform(TargetPlatform.ios, buildNameToParse);
+  }
+
+  if (buildNameIsMissing && buildNumberIsMissing) {
```

Comment:
Is is useful to handle this separately? It seems like making the below non-`else` `if`s and just having two lines of output would be as easy to understand, and less code.

Reply:
I had that at first because it was much easier to write, but I decided to combine it to give a nicer message.  Here's the difference in output.  Current code:
```shell
Signing iOS app for device deployment using developer identity: "..."
Warning: Missing build name (CFBundleShortVersionString), defaulting to 1.0.0.
Missing build number (CFBundleVersion), defaulting to 1.
Set a build name and number in the pubspec.yaml version before submitting to the App Store.
```
![Screen Shot 2019-09-16 at 5 20 52 PM](https://user-images.githubusercontent.com/682784/65001913-5d1f1680-d8a6-11e9-9ea7-3c2f00f8c885.png)

vs your suggestion:
```
Signing iOS app for device deployment using developer identity: "..."
Warning: Missing build name (CFBundleShortVersionString), defaulting to 1.0.0.
Set a build name in the pubspec.yaml version before submitting to the App Store.
Warning: Missing build number (CFBundleVersion), defaulting to 1.
Set a build number in the pubspec.yaml version before submitting to the App Store.
```
![Screen Shot 2019-09-16 at 5 25 28 PM](https://user-images.githubusercontent.com/682784/65002099-0d8d1a80-d8a7-11e9-85b6-baf9730ebc74.png)

What do you think?

Reply:
I actually prefer the second; I think it makes it clearer that there are two distinct issues that need to be resolved. It's clearly subjective though, so your call.

Reply:
@zanderso Do you have a preference?

Reply:
How about:
```dart
if (buildNameIsMissing) {
  printError('Warning: Missing build name (CFBundleShortVersionString), defaulting to $defaultBuildName.');
}
if (buildNumberIsMissing) {
  printError('Warning: Missing build number (CFBundleVersion), defaulting to $defaultBuildNumber.\n');
}
if (buildNameIsMissing || buildNumberIsMissing) {
  printError('Action Required: You must set a build name and number in the pubspec.yaml '
             'file before submitting to the App Store.');
}
```

Reply:
Sounds good.  I added "version field" just so they knew how to add the name and number:
```
    printError('Action Required: You must set a build name and number in the pubspec.yaml '
               'file version field before submitting to the App Store.');
```